src/rollup/RollupCreator.sol

```diff
@@ -89,22 +100,36 @@ contract RollupCreator is Ownable {
         address _batchPoster,
         address[] calldata _validators,
         address _nativeToken
-    ) external returns (address) {
+    ) external payable returns (address) {
+        return createRollup(config, _batchPoster, _validators, _nativeToken, true);
+    }
+
+    /**
+     * @notice Create a new rollup
+     * @dev After this setup:
+     * @dev - Rollup should be the owner of bridge
+     * @dev - RollupOwner should be the owner of Rollup's ProxyAdmin
+     * @dev - RollupOwner should be the owner of Rollup
+     * @dev - Bridge should have a single inbox and outbox
+     * @dev - Validators and batch poster should be set if provided
+     * @param config       The configuration for the rollup
+     * @param _batchPoster The address of the batch poster, not used when set to zero address
+     * @param _validators  The list of validator addresses, not used when set to empty list
+     * @param _nativeToken Address of the custom fee token used by rollup. If rollup is ETH-based address(0) should be provided
+     * @param _deployFactoriesToL2 Whether to deploy L2 factories using retryable tickets. If true, retryables need to be paid for in native currency.
```

Comment:
I think we should include more comments here about the consequences of not deploying the factories

Reply:
https://github.com/OffchainLabs/nitro-contracts/pull/59/commits/116d6515349f01736b67ba3870ac6535d08150fd

Reply:
Isnt there also the risk that someone else, other than the original deployer, could send these transactions and burn the nonce without

Reply:
yep, I mean it doesn't matter who triggers the transaction, problem is if it fails and nonce gets burned. I updated comment to be more clear about that.